lock-rewards/src/lib.rs

```diff
@@ -0,0 +1,200 @@
+#![no_std]
+
+elrond_wasm::imports!();
+
+pub mod user_deposit;
+use user_deposit::*;
+
+#[elrond_wasm_derive::contract]
+pub trait LockRewards {
+    #[init]
+    fn init(
+        &self,
+        stablecoin_token_id: TokenIdentifier,
+        percentage_reward_per_block: Self::BigUint,
+    ) -> SCResult<()> {
+        require!(
+            stablecoin_token_id.is_valid_esdt_identifier(),
+            "invalid stablecoin token id"
+        );
+
+        self.try_set_percentage_rewards_per_block(&percentage_reward_per_block)
+    }
+
+    // endpoints - owner-only
+
+    fn set_percentage_reward_per_block(
```

Comment:
Comment says it's an endpoint, and the `only_owner!` check is there, but the function is not marked as endpoint.
```suggestion
    // endpoints - owner-only

    #[endpoint(setPercentageRewardPerBlock)]
    fn set_percentage_reward_per_block(
```